contracts/interfaces/IFractalUsul.sol

```diff
@@ -2,44 +2,161 @@
 pragma solidity ^0.8.0;
 
 import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import "../FractalUsul.sol";
 
 interface IFractalUsul {
-    /// @dev Returns the hash of a transaction in a proposal.
-    /// @param proposalId the proposal to inspect.
-    /// @param index the transaction to inspect.
-    /// @return transaction hash.
-    function getTxHash(uint256 proposalId, uint256 index)
-        external
-        view
-        returns (bytes32);
+    enum ProposalState {
+        ACTIVE,
+        TIMELOCKED,
+        EXECUTABLE,
+        EXECUTED
```

Comment:
does this need a `FAILED` state of some sort?

If doesn't pass it doesn't make sense to stay in the ACTIVE state

Reply:
we discussed this this morning,

it's possible some strategies wouldn't have the concept of quorum, so it's probably best to have a single "FAILED" state, and let the front end figure out why

Reply:
Cool, just pushed up changes for adding the FAILED state

Reply:
sick